@mui/material@5.8.6 includes incorrect .d.ts files

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

https://unpkg.com/@mui/material@5.8.6/styles/CssVarsProvider.d.ts

<img width="1158" alt="image" src="https://user-images.githubusercontent.com/24715727/176158268-08213b98-17e5-4ad6-b7f9-ccf1f111bef2.png">


### Expected behavior 🤔

https://unpkg.com/@mui/material@5.8.5/styles/CssVarsProvider.d.ts

<img width="1006" alt="image" src="https://user-images.githubusercontent.com/24715727/176158305-9d579998-5e79-438f-bb4d-ab98019c0dcb.png">


### Steps to reproduce 🕹

Steps:

1. Update MUI to the latest 5.8.6 version
2. Run `tsc -b` 
3. Got error `error TS2307: Cannot find module 'packages/mui-system/src/cssVars/createCssVarsProvider' or its corresponding type declarations.`



### Context 🔦

N/A

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 12.4
  Binaries:
    Node: 16.15.0 - /opt/homebrew/opt/node@16/bin/node
    Yarn: Not Found
    npm: 8.5.5 - /opt/homebrew/opt/node@16/bin/npm
  Browsers:
    Chrome: 103.0.5060.53
    Edge: Not Found
    Firefox: 101.0.1
    Safari: 15.5
  npmPackages:
    typescript: ~4.7.3 => 4.7.3
```
</details>

I am receiving the same error on my GitHub Action Runner:

```
 > tsc
node_modules/@mui/material/styles/CssVarsProvider.d.ts:3:102 - error TS2307: Cannot find module 'packages/mui-system/src/cssVars/createCssVarsProvider' or its corresponding type declarations.
3 declare const Experimental_CssVarsProvider: (props: import("react").PropsWithChildren<Partial<import("packages/mui-system/src/cssVars/createCssVarsProvider").CssVarsProviderConfig<SupportedColorScheme>> & {
                                                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@mui/material/styles/CssVarsProvider.d.ts:9:125 - error TS2307: Cannot find module 'packages/mui-system/src/cssVars/createCssVarsProvider' or its corresponding type declarations.
9 }>) => import("react").ReactElement<any, string | import("react").JSXElementConstructor<any>>, useColorScheme: () => import("packages/mui-system/src/cssVars/createCssVarsProvider").ColorSchemeContextValue<SupportedColorScheme>, getInitColorSchemeScript: typeof import("packages/mui-system/src/cssVars/getInitColorSchemeScript").default;
                                                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@mui/material/styles/CssVarsProvider.d.ts:9:269 - error TS2307: Cannot find module 'packages/mui-system/src/cssVars/getInitColorSchemeScript' or its corresponding type declarations.
9 }>) => import("react").ReactElement<any, string | import("react").JSXElementConstructor<any>>, useColorScheme: () => import("packages/mui-system/src/cssVars/createCssVarsProvider").ColorSchemeContextValue<SupportedColorScheme>, getInitColorSchemeScript: typeof import("packages/mui-system/src/cssVars/getInitColorSchemeScript").default;
                                                                                                                                                                                                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 3 errors in the same file, starting at: node_modules/@mui/material/styles/CssVarsProvider.d.ts:3
```

The `@mui/material` version is listed as `^5.6.3` in my `package.json` file.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -38,6 +38,7 @@ jobs:
       - run: yarn release:changelog
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: yarn validate-declarations
       - name: yarn release:tag
         run: |
           git remote -v

--- a/package.json
+++ b/package.json
@@ -67,7 +67,8 @@
     "test:unit": "cross-env NODE_ENV=test mocha 'packages/**/*.test.{js,ts,tsx}' 'docs/**/*.test.{js,ts,tsx}' 'scripts/**/*.test.{js,ts,tsx}' 'test/utils/**/*.test.{js,ts,tsx}'",
     "test:argos": "node ./scripts/pushArgos.js",
     "typescript": "lerna run --no-bail --parallel typescript",
-    "typescript:ci": "lerna run --concurrency 7 --no-bail --no-sort typescript"
+    "typescript:ci": "lerna run --concurrency 7 --no-bail --no-sort typescript",
+    "validate-declarations": "babel-node --extensions \".ts\" scripts/validateTypescriptDeclarations.ts"
   },
   "devDependencies": {
     "@babel/cli": "^7.17.6",

--- a/packages/mui-material/src/styles/CssVarsProvider.tsx
+++ b/packages/mui-material/src/styles/CssVarsProvider.tsx
@@ -1,4 +1,3 @@
-/* eslint-disable @typescript-eslint/naming-convention */
 import { unstable_createCssVarsProvider as createCssVarsProvider } from '@mui/system';
 import experimental_extendTheme, {
   SupportedColorScheme,
@@ -8,11 +7,10 @@ import createTypography from './createTypography';
 
 const defaultTheme = experimental_extendTheme();
 
-const {
-  CssVarsProvider: Experimental_CssVarsProvider,
-  useColorScheme,
-  getInitColorSchemeScript,
-} = createCssVarsProvider<SupportedColorScheme, CssVarsTheme>({
+const { CssVarsProvider, useColorScheme, getInitColorSchemeScript } = createCssVarsProvider<
+  SupportedColorScheme,
+  CssVarsTheme
+>({
   theme: defaultTheme,
   attribute: 'data-mui-color-scheme',
   modeStorageKey: 'mui-mode',
@@ -35,4 +33,8 @@ const {
     (keys[0] === 'palette' && !!keys[1]?.match(/(mode|contrastThreshold|tonalOffset)/)),
 });
 
-export { useColorScheme, getInitColorSchemeScript, Experimental_CssVarsProvider };
+export {
+  useColorScheme,
+  getInitColorSchemeScript,
+  CssVarsProvider as Experimental_CssVarsProvider,
+};

--- a/packages/mui-system/src/cssVars/index.ts
+++ b/packages/mui-system/src/cssVars/index.ts
@@ -1,2 +1,8 @@
 export { default } from './createCssVarsProvider';
-export type { CreateCssVarsProviderResult } from './createCssVarsProvider';
+export type {
+  CreateCssVarsProviderResult,
+  CssVarsProviderConfig,
+  ColorSchemeContextValue,
+} from './createCssVarsProvider';
+
+export { default as getInitColorSchemeScript } from './getInitColorSchemeScript';

--- a/scripts/validateTypescriptDeclarations.ts
+++ b/scripts/validateTypescriptDeclarations.ts
@@ -1,0 +1,26 @@
+/* eslint-disable no-console */
+import globby from 'globby';
+import fs from 'fs';
+
+/**
+ * Validates if there are no missing exports from TS files that would
+ * result in an import from a local file.
+ */
+function validateFiles() {
+  const declarationFiles = globby.sync(['packages/*/build/**/*.d.ts']);
+  const invalidFiles = declarationFiles.filter((file) => {
+    const content = fs.readFileSync(file, 'utf8');
+    const regex = /import\(["']packages\//gm;
+    return regex.test(content);
+  });
+
+  if (invalidFiles.length > 0) {
+    console.error('Found invalid imports in the following files:');
+    invalidFiles.forEach((file) => console.error(file));
+    process.exit(1);
+  }
+
+  console.log('Found no invalid import statements in built declaration files.');
+}
+
+validateFiles();

